plugins/thunderbolt/fu-thunderbolt-retimer.c

```diff
@@ -20,37 +20,20 @@ G_DEFINE_TYPE(FuThunderboltRetimer, fu_thunderbolt_retimer, FU_TYPE_THUNDERBOLT_
 static FuUdevDevice *
 fu_thunderbolt_retimer_get_udev_grandparent(FuDevice *device, GError **error)
 {
-	g_autoptr(GUdevDevice) udev_parent1 = NULL;
-	g_autoptr(GUdevDevice) udev_parent2 = NULL;
-	GUdevDevice *udev_device = NULL;
 	FuThunderboltRetimer *self = FU_THUNDERBOLT_RETIMER(device);
+	g_autoptr(FuUdevDevice) udev_parent1 = NULL;
+	g_autoptr(FuUdevDevice) udev_parent2 = NULL;
 
-	udev_device = fu_udev_device_get_dev(FU_UDEV_DEVICE(device));
-	if (udev_device == NULL) {
-		g_set_error_literal(error,
-				    FWUPD_ERROR,
-				    FWUPD_ERROR_INTERNAL,
-				    "failed to get udev device for retimer");
+	udev_parent1 =
+	    fu_udev_device_get_parent_with_subsystem(FU_UDEV_DEVICE(self), NULL, NULL, error);
```

Comment:
shouldn't the second arg be something like `"thunderbolt"`?

Reply:
Yes, but that's `Subject: [PATCH 03/13] trivial: Show what the subsystem should have been when compiling without SUPPORTED_BUILD` in the next PR :)

Reply:
I've merged in the important bits now.